src/analysis/calculus/inverse.lean

```diff
@@ -0,0 +1,582 @@
+def to_local_homeomorph (hs : is_open s) : local_homeomorph E F :=
+{ to_local_equiv := hf.to_local_equiv hc,
+    open_source := hs,
+    open_target :=
+      begin
+        cases hc with hE hc,
+        { resetI,
+          haveI hF : subsingleton F := f'.to_linear_equiv.to_equiv.symm.subsingleton,
+          apply is_open_discrete },
+        change is_open (f '' s),
+        simp only [is_open_iff_mem_nhds, nhds_basis_closed_ball.mem_iff, ball_image_iff] at hs ⊢,
+        intros x hx,
+        rcases hs x hx with ⟨ε, ε0, hε⟩,
+        refine ⟨(N⁻¹ - c) * ε, mul_pos (sub_pos.2 hc) ε0, _⟩,
+        exact (hf.surj_on_closed_ball (or.inr hc) (le_of_lt ε0) hε).mono hε (subset.refl _)
+      end,
+    continuous_to_fun := hf.continuous_on,
+    continuous_inv_fun := hf.inverse_continuous_on hc }
+
+end
+
+@[simp] lemma to_local_homeomorph_to_fun (hs : is_open s) :
+  (hf.to_local_homeomorph f s hc hs).to_fun = f := rfl
+
+@[simp] lemma to_local_homeomorph_source (hs : is_open s) :
+  (hf.to_local_homeomorph f s hc hs).source = s := rfl
+
+@[simp] lemma to_local_homeomorph_target (hs : is_open s) :
+  (hf.to_local_homeomorph f s hc hs).target = f '' s := rfl
+
+lemma closed_ball_subset_target (hs : is_open s) {b : E} (ε0 : 0 ≤ ε) (hε : closed_ball b ε ⊆ s) :
+  closed_ball (f b) ((N⁻¹ - c) * ε) ⊆ (hf.to_local_homeomorph f s hc hs).target :=
+(hf.surj_on_closed_ball hc ε0 hε).mono hε (subset.refl _)
+
+end approximates_linear_on
+
+namespace has_strict_fderiv_at
+
+section
+variables {f : E → F} {f' : E →L[𝕜] F} {a : E}
+
+/-- If `f` has derivative `f'` at `a` in strict sense and `c > 0`, then `f` approximates `f'`
+with constant `c` on some neighborhood of `a`. -/
+lemma approximates_deriv_on_nhds {f : E → F} {f' : E →L[𝕜] F} {a : E}
+  (hf : has_strict_fderiv_at f f' a) {c : ℝ≥0} (hc : subsingleton E ∨ 0 < c) :
+  ∃ s ∈ 𝓝 a, approximates_linear_on f f' s c :=
+begin
+  cases hc with hE hc,
+  { refine ⟨univ, mem_nhds_sets is_open_univ trivial, λ x hx y hy, _⟩,
+    simp [@subsingleton.elim E hE x y] },
+  have := hf.def hc,
+  rw [nhds_prod_eq, filter.eventually, mem_prod_same_iff] at this,
+  rcases this with ⟨s, has, hs⟩,
+  exact ⟨s, has, λ x hx y hy, hs (mk_mem_prod hx hy)⟩
+end
+
+end
+
+variables [cs : complete_space E] {f : E → F} (f' : E ≃L[𝕜] F) {a : E}
+  (hf : has_strict_fderiv_at f (f' : E →L[𝕜] F) a)
+
+variable {f'}
+include hf
+
+lemma approximates_deriv_on_open_nhds :
+  ∃ (s : set E) (hs :  a ∈ s ∧ is_open s),
+    approximates_linear_on f (f' : E →L[𝕜] F) s ((nnnorm (f'.symm : F →L[𝕜] E))⁻¹ / 2) :=
+begin
+  refine ((nhds_basis_opens a).exists_iff _).1 _,
+  exact (λ s t, approximates_linear_on.mono_set),
+  exact (hf.approximates_deriv_on_nhds $ f'.subsingleton_or_nnnorm_symm_pos.imp id $
+    λ hf', nnreal.half_pos $ nnreal.inv_pos.2 $ hf')
+end
+
+include cs
+
+variable (f)
+
+/-- Given a function with a bijective strict derivative at `a`, returns a `local_homeomorph`
```

Comment:
```suggestion
/-- Given a function with an invertible strict derivative at `a`, returns a `local_homeomorph`
```
`bijective` is a little bit too weak, you really want the inverse to be continuous (which is what you assume). Replacing it with `invertible` is better, because you are not saying in which category you are requiring invertibility.